contract.md

```diff
@@ -12,11 +16,19 @@ like to work on, as part of your project (for example, other things related to D
 you have come across, or topics from other classes)? Why are you interested in working on
 these things?_
 
+I would like to work on external DSLs. While we have covered these conceptually in class, I found myself more challenged understanding how they work and what it would look like to implement one. Even for the PicoBot assignment, I experienced many difficulties so I felt pressured by time to use an internal DSL as that is what I felt I had a clearer understanding on and that I could confidently work with. 
+
+Thus, I am somewhat biased towards challenging myself to take on an external DSL approach for this project so that I can ensure I understand what that implementation entails. Since I had difficulties understanding how external DSLs work, I think that by implementing one I will be able to really clear up my understanding of external DSLs. However, I will first need to review the lessons on external DSLs, such as "Programs, interpreters, and compilers", to clear up any background information I will need going into beginning my implementation.
+
 ## Time management plan
 
 _How do you plan to set aside time outside of class, to work on the project? Are there
 intermediate milestones that you can create, to help you make consistent progress?_
 
+I plan to set aside time on Wednesday afternoons to work on the project. I think this will be useful because it will come after our studio sessions on Wednesday and I have been experiencing more issues with context switching (jumping from one class' work to another) recently so by having longer continuous blocks of time dedicated to working on the project I will have less issues with getting started or worrying about other assignments. Additionally, I do not have classes after this or assignments due usually after Tuesdays, just a meeting in the late evening so I should be free of other time-sensitive pressures during these periods.
+
+I think that the intermediate milestones will be very dependent on how I get started but for my short term goals I would like to have an implementation by the end of this upcoming week (end of 3/31) so that way I can begin the process of designing the DSL in a general purpose programming language and my ideal example in my DSL by the Critique day the following week (start of class 4/3). Relating this back to the project planning discussed in class, this would be the start of the two ends of the bridge that will connect my ideal design to the current method of writing it and closing the gap between the two.
+
```

Comment:
I think this is a good plan. I might also add that mapping out your idea of a language this being:
-here is a list of function that the user should have/would be nice to have
->here is my end goal
building something like this may be to your benefit and scope your ideas through your implementation.